cpp/src/arrow/array/array_base.h

```diff
@@ -69,16 +69,37 @@ class ARROW_EXPORT Array {
     // a potential inner-branch removal.
     if (type_id() == Type::SPARSE_UNION) {
       return !internal::IsNullSparseUnion(*data_, i);
-    }
```

Comment:
Why change it to if else? I think it's not necessary here
clang-tidy has a check about this style ( https://clang.llvm.org/extra/clang-tidy/checks/readability/else-after-return.html ), although we don't enable this check, I think just return without if else is ok

Reply:
More compact. In sync with the `if constexpr` used in the other cases. Arrow's `.clang-tidy` doesn't complain about this. No extra indentation levels are introduced because of this.